src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2492 @@
+        pixel_mask: Optional[Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> MaskFormerForInstanceSegmentationOutput:
+        r"""
+        mask_labels (`torch.FloatTensor`, *optional*):
+            The target mask of shape `(num_classes, height, width)`.
+        class_labels (`torch.LongTensor`, *optional*):
+            The target labels of shape `(num_classes)`.
+
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import MaskFormerFeatureExtractor, MaskFormerForInstanceSegmentation
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> feature_extractor = MaskFormerFeatureExtractor.from_pretrained("facebook/maskformer-swin-base-ade")
+        >>> inputs = feature_extractor(images=image, return_tensors="pt")
+
+        >>> model = MaskFormerForInstanceSegmentation.from_pretrained("facebook/maskformer-swin-base-ade")
+        >>> outputs = model(**inputs)
+        >>> # model predicts class_queries_logits of shape `(batch_size, num_queries)`
+        >>> # and masks_queries_logits of shape `(batch_size, num_queries, height, width)`
+        >>> class_queries_logits = outputs.class_queries_logits
+        >>> masks_queries_logits = outputs.masks_queries_logits
+
+        >>> # you can pass them to feature_extractor for postprocessing
+        >>> output = feature_extractor.post_process_segmentation(outputs)
+        >>> output = feature_extractor.post_process_semantic_segmentation(outputs)
+
+        >>> output = feature_extractor.post_process_panoptic_segmentation(outputs)
+        ```
+        """
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs: MaskFormerModelOutput = self.model(
+            pixel_values,
+            pixel_mask,
+            output_hidden_states=output_hidden_states,
+            return_dict=True,
+            output_attentions=output_attentions,
+        )
+
+        loss, loss_dict, auxilary_logits = None, None, None
+
+        class_queries_logits, masks_queries_logits, auxilary_logits = self.get_logits(outputs)
+
+        we_have_labels = mask_labels is not None and class_labels is not None
+
+        if we_have_labels:
+            loss_dict: Dict[str, Tensor] = self.get_loss_dict(
+                masks_queries_logits, class_queries_logits, mask_labels, class_labels, auxilary_logits
+            )
+            loss = self.get_loss(loss_dict)
```

Comment:
I made a comment here on the previous PR which I'd like to see addressed.

I'd do:

```
loss = None
if mask_labels is not None and class_labels is not None:
   # compute loss
```
as this is done in all other models.